src/errors.rs

```diff
@@ -1,4 +1,4 @@
-#![allow(bare_trait_object)] // FIXME -- error-chain uses old-style trait objects
+#![allow(bare_trait_objects)] // FIXME -- error-chain uses old-style trait objects
```

Comment:
This change has nothing to do with the refactor. The warning was just bothering me and I forgot to exclude it from my commit.